crates/biome_js_analyze/src/analyzers/correctness/no_self_assign.rs

```diff
@@ -713,7 +713,7 @@ impl TryFrom<(AnyNameLike, AnyNameLike)> for IdentifiersLike {
                 AnyNameLike::AnyJsLiteralExpression(right),
             ) => Ok(Self::Literal(left, right)),
 
-            _ => unreachable!("you should map the correct references"),
+            _ => Err(()),
```

Comment:
the `unreachable` was there for a **reason**, which means we haven't mapped a specific type that we should

Reply:
I don't think so. Otherwise, why not implement `From`? Moreover, all branches of the match checks that the two values have the same type. This is a wrong assumption.